### PR TITLE
ci-arm: CLI-only docs and per-checkout container isolation fix

### PR DESCRIPTION
**Pull request by midleman**
### Summary
Documents driving the ci-arm CI lab headlessly (no VS Code UI), restructures the README by audience, and fixes a real container-collision bug found while using it this session.
- Adds Claude Workflows (CLI-only) steps: env setup, branch switching, bring-up, cold/warm/hot build detection, running tests
- Splits docs into User Workflows (VS Code UI) / Claude Workflows
- Fixes `initialize.sh` to pin `COMPOSE_PROJECT_NAME` per checkout -- two checkouts previously shared one set of containers/volumes silently instead of erroring

### QA Notes
Docs + script changes, verified end-to-end: fresh worktree build, e2e test run (search.test.ts) under the new isolated container project, and a live container-collision repro before/after the fix.

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -19,6 +19,18 @@ Edge cases:
 
 **Exception for vitest test files:** `npm run build-check` excludes `*.vitest.ts` files, so it won't surface type errors in tests. To check those, run `npm run test:positron:check-ts` (wraps `tsc --noEmit -p ./vitest.tsconfig.json --skipLibCheck`). It is scoped to vitest files + their ambient declarations and completes in seconds. The output matches what the VS Code Problems pane shows for `.vitest.{ts,tsx}` files. Filter to a specific file with: `npm run test:positron:check-ts 2>&1 | grep '<file>.vitest.ts'`.
 
+### Reproducing CI Failures Locally (Posit-internal)
+
+Only when the user explicitly asks to reproduce a CI-only failure on the real CI
+image (arm64). Requires private GHCR access + license files (not for external
+contributors) -- gate on `gh api repos/posit-dev/positron-ci-images` (404 = no
+access; tell the user, don't attempt setup).
```
Comment (midleman) [inline CLAUDE.md:27]: @testlabauto aren't the images moving into this repo? will they be protected? or does this gate completely go away? just something i'll need to update....

Comment (testlabauto) [reply CLAUDE.md:27]: Yep. I will merge it in shortly.

**Review by testlabauto — APPROVED** [review]
Personally, I think that Claude driving such a time consuming process for the user might be strange, but lets give it a try!
